File: src/main/java/com/paragon/client/systems/module/impl/misc/ChatModifications.java
package com.paragon.client.systems.module.impl.misc;

import com.paragon.Paragon;
import com.paragon.api.module.Module;
import com.paragon.api.module.Category;
import com.paragon.api.setting.Setting;
import net.minecraftforge.client.event.ClientChatEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

/**
 * @author Wolfsurge
 */
public class ChatModifications extends Module {

    public static ChatModifications INSTANCE;

    public static Setting<Boolean> coloured = new Setting<>("Coloured", false)
            .setDescription("Adds a '>' before the message");

    public static Setting<Boolean> suffix = new Setting<>("Suffix", true)
            .setDescription("Adds a Paragon suffix to the end of the message");

    public ChatModifications() {
        super("ChatModifications", Category.MISC, "Changes the way you send messages");

        INSTANCE = this;
    }

    @SubscribeEvent
    public void onChat(ClientChatEvent event) {
        if (Paragon.INSTANCE.getCommandManager().startsWithPrefix(event.getMessage())) {
            return;
        }

        if (coloured.getValue()) {
            event.setMessage("> " + event.getMessage());
        }

        if (suffix.getValue()) {
            event.setMessage(event.getMessage() + " | Paragon");
        }
    }

}

File: src/main/kotlin/com/paragon/client/systems/module/impl/misc/ChatModifications.kt
package com.paragon.client.systems.module.impl.misc

import com.paragon.Paragon
import com.paragon.api.module.Category
import com.paragon.api.module.Module
import com.paragon.api.setting.Setting
import net.minecraftforge.client.event.ClientChatEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

/**
 * @author Surge
 */
object ChatModifications : Module("ChatModifications", Category.MISC, "Changes the way you send messages") {

    private val coloured = Setting(
        "Coloured",
        false
    ) describedBy "Adds a '>' before the message"

    private val suffix = Setting(
        "Suffix",
        true
    ) describedBy "Adds a Paragon suffix to the end of the message"

    @SubscribeEvent
    fun onChat(event: ClientChatEvent) {
        if (event.message.startsWith("/") || Paragon.INSTANCE.commandManager.startsWithPrefix(event.message)) {
            return
        }

        if (coloured.value) {
            event.message = "> " + event.message
        }

        if (suffix.value) {
            event.message = event.message + " | Paragon"
        }
    }

}